junit-platform-engine/src/main/java/org/junit/platform/engine/support/hierarchical/HierarchicalTestExecutor.java

```diff
@@ -158,16 +158,21 @@ private void cleanUp() {
 		private void reportDone() {
 			if (executionResult != null) {
 				addExecutionErrorsToTestExecutionResult();
+				node.nodeFinished(context, testDescriptor, executionResult);
```

Comment:
Must be wrapped in a try-catch block in order to ensure that a `TestWatcher` that throws an exception cannot abort the execution and reporting.

Any exception should be logged (e.g., at `WARNING` log level) and swallowed.